dali/operators/audio/nonsilence_op.cc

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/core/static_switch.h"
+#include "dali/operators/audio/nonsilence_op.h"
+#include "dali/pipeline/data/views.h"
+
+namespace dali {
+
+DALI_SCHEMA(NonsilenceRegion)
+                .DocStr(R"code(The operator performs leading and trailing silence detection in an audio buffer.
+This operators' behaviour can be described as::
+
+  def nonsilence(buffer, cutoff_value, top_db, ref_db):
+      buffer = ToDecibels(buffer, top_db, ref_db)
+      begin = end = 0
+      for i in [0 .. len(buffer)):
+          if buffer[i] > cutoff_value:
+              begin = i; break
+      for i in [len(buffer)-1 .. begin]:
+          if buffer[i] > cutoff_value:
+              end = i; break
+      length = end - begin + 1
+      return begin, length
+
+
+
++----------------------+-----------------------------------------+
+| Inputs               | Outputs                                 |
++=======+==============+=======+=================================+
+|       | | 1-D audio  | **0** | Begin index of nonsilent region |
+| **0** | | buffer     +-------+---------------------------------+
+|       |              | **1** | Length of nonsilent region      |
++-------+--------------+-------+---------------------------------+
+
+If ``Outputs[1] == 0``, ``Outputs[0]`` value is undefined
+)code")
+                .NumInput(1)
+                .NumOutput(detail::kNumOutputs)
+                .AddOptionalArg("cutoff_db",
+                                R"code(The threshold [dB], below which everything is considered as silence)code",
+                                60.f)
+                .AddOptionalArg("window_length", R"code(Size of a sliding window)code", 2048)
```

Comment:
Again, we should be more explicit. What is this window used for. E.g. "Size of the sliding window used to calculate RMS..." or whatever it is